libs/server/PubSub/SubscribeBroker.cs

```diff
@@ -408,6 +410,164 @@ public unsafe void Publish(byte* key, byte* value, int valueLength, bool ascii =
             log.Enqueue(logEntryBytes);
         }
 
+        /// <summary>
+        /// Retrieves the collection of channels that have active subscriptions.
+        /// </summary>
+        /// <returns>The collection of channels.</returns>
+        public unsafe void Channels(ref ObjectInput input, ref SpanByteAndMemory output)
+        {
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var ptr = output.SpanByte.ToPointer();
+
+            var curr = ptr;
+            var end = curr + output.Length;
+
+            try
+            {
+                if (subscriptions is null || subscriptions.Count == 0)
+                {
+                    while (!RespWriteUtils.WriteEmptyArray(ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                    return;
+                }
+
+                if (input.parseState.Count == 0)
+                {
+                    while (!RespWriteUtils.WriteArrayLength(subscriptions.Count, ref curr, end))
+                        ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+
+                    foreach (var key in subscriptions.Keys)
+                    {
+                        while (!RespWriteUtils.WriteSimpleString(key.AsSpan().Slice(sizeof(int)), ref curr, end))
+                            ObjectUtils.ReallocateOutput(ref output, ref isMemory, ref ptr, ref ptrHandle, ref curr, ref end);
+                    }
+                    return;
+                }
+
+                var totalArrayHeaderLen = 0;
+                while (!RespWriteUtils.WriteArrayLength(subscriptions.Count, ref curr, end, out var _, out totalArrayHeaderLen))
```

Comment:
Gotcha... This needs some comments to explain what you're doing here. Code is hard to read as-is...

Reply:
Please put a comment here, and below where you do the size adjustments. And mention why you are doing it this way, i.e., to avoid two passes over the subscriptions etc.

Reply:
Done